src/main/LaunchInterceptorConditions.java

```diff
@@ -40,22 +42,102 @@ public static boolean ConditionThree(double AREA1, int[][] POINTS) {
-        // do something
-        return true;
+        /* The idea is that numpoints is POINTS.length */
+        if(QPTS < 2 || QPTS > POINTS.length){
+            return false;
+        }
+    
+        if(QUADS < 1 || QUADS > 3){
+            return false;
+        }
+
+        for(int i = 0; i < POINTS.length - QPTS -1; i++){
+            boolean[] FoundQuads = {false, false, false, false};
+            for(int j = 0; i < QPTS; j++){
+                if(POINTS[i][0] >= 0 && POINTS[i][1] >= 0){
+                    FoundQuads[0] = true;
+                }
+                if(POINTS[i][0] < 0 && POINTS[i][1] >= 0){
+                    FoundQuads[1] = true;
+                }
+                if(POINTS[i][0] <= 0 && POINTS[i][1] < 0){
+                    FoundQuads[2] = true;
+                }
+                if(POINTS[i][0] > 0 && POINTS[i][1] < 0){
+                    FoundQuads[3] = true;
+                }
+            }
+
+            int QuadsWithPoints = 0;
+            for(int l = 0; l < FoundQuads.length; l++){
+                if(FoundQuads[i]){
+                    QuadsWithPoints++;
+                }
+            }
+            if(QuadsWithPoints > QUADS){
+                return true;
+            }
+        }
+
+        return false;
     }
 
     public static boolean ConditionFive(int[][] POINTS) {
-        // do something
-        return true;
+        for(int i = 1; i < POINTS.length; i++){
+            if(POINTS[i][0] - POINTS[i-1][0] < 0){
+                return true;
+            }
+        }
+        return false;
     }
 
     public static boolean ConditionSix(int NPTS, double DIST, int NUMPOINTS, int[][] POINTS) {
-        // do something
-        return true;
+        if(3 > NPTS || NPTS > NUMPOINTS){
+            return false;
+        }
+        if(0 > DIST){
+            return false;
+        }
+        for(int i = 0; i < POINTS.length - NPTS; i++){
+            // First loop changes the data set moving it from start of NUMPOINTS to end.
+            if(POINTS[i][0] == POINTS[i + NPTS][0] && POINTS[i][1] == POINTS[i + NPTS][1]){
+                // checking to see if start and end points are the same
+                
+                for(int j = i + 1;j < i + NPTS - 1; j++  ){ // i is the start point so start on +1 
+                    //-1 so that we don't waste time on end point
+            // Second loop iterates through every point checking distance compared to DIST.
+                    if(HelperFunction.calculateDistance(POINTS[i], POINTS[j]) > DIST){
+                        return true;
+                    }
+                }
+            }
+            else{
+                for(int j = i + 1; j < i + NPTS; j++){
+                    int[] MidwayPoint = {(POINTS[i][0]-POINTS[i + NPTS][0])/2, (POINTS[i][1]-POINTS[i + NPTS][1])/2};
+                    if(HelperFunction.calculateDistance(MidwayPoint, POINTS[j]) > DIST){
```

Comment:
This is not the property which is to be checked. The spec says that the distance from the line connecting these two points is to be greater than DIST. Here you have distance from midpoint.

Reply:
You can use a [formula](https://en.wikipedia.org/wiki/Distance_from_a_point_to_a_line#Line_defined_by_two_points) from Wikipedia to calculate the distance.